### PR TITLE
UI support for streaming CSV export to the browser

### PR DESCRIPTION
**Pull request by kroepke**
requires https://github.com/Graylog2/graylog2-server/pull/1149

issue #765

### CODE REVIEW

File: app/controllers/SearchController.java [hunk 1]
```diff
@@ -253,16 +252,16 @@ public Result exportAsCsv(String q, String filter, String rangeType, int relativ
...
-            String message = "There was a problem with your search. We expected HTTP 200, but got a HTTP " + e.getHttpCode() + ".";
-            return status(504, views.html.errors.error.render(message, e, request()));
-        }
+            //s = search.searchAsCsv(selectedFields);
+//        } catch (IOException e) {
+//            return status(504, views.html.errors.error.render(ApiClient.ERROR_MSG_IO, e, request()));
+//        } catch (APIException e) {
+//            String message = "There was a problem with your search. We expected HTTP 200, but got a HTTP " + e.getHttpCode() + ".";
+//            return status(504, views.html.errors.error.render(message, e, request()));
+//        }
```
Comment (bernd) [inline app/controllers/SearchController.java]: I guess that commented code can be removed?

File: app/controllers/SearchController.java [hunk 2]
```diff
@@ -253,16 +252,16 @@ public Result exportAsCsv(String q, String filter, String rangeType, int relativ
...
 
         // TODO streaming the result
```
Comment (bernd) [inline app/controllers/SearchController.java]: This TODO is obsolete now? ;)

File: javascript/src/stores/search/SearchStore.ts [hunk 3]
```diff
@@ -238,6 +238,13 @@ class SearchStore {
         searchURLParams = searchURLParams.set(param, value);
         URLUtils.openLink(jsRoutes.controllers.SearchControllerV2.index().url + "?" + Qs.stringify(searchURLParams.toJS()));
     }
+
+    getCsvExportURL(): string {
+        var searchURLParams = this.getOriginalSearchURLParams();
+        searchURLParams = searchURLParams.delete('page');
+        // function (q,filter,rangetype,relative,from,to,keyword,fields)
```
Comment (bernd) [inline javascript/src/stores/search/SearchStore.ts]: I don't know what this comment means.

Comment (kroepke) [inline javascript/src/stores/search/SearchStore.ts]: It was left over, I'll remove it. It's the signature for the javascript router for that controller action.

**Comment by bernd** [comment]
Besides my comments, LGTM.

File: app/controllers/SearchController.java (hunk 1 above)
Comment (kroepke) [inline app/controllers/SearchController.java]: yeah, but we'll remove the entire old controller once we move the v2 back :)

**Pull request merged**